### PR TITLE
[services] indicate how many errors we have seen

### PR DESCRIPTION
**Pull request by danking**
We are increasingly seeing errors from "Connection reset" which we switched from "transient" to "retry once". The current code makes it impossible to determine if we are correctly retrying this error once.

If we see that there are a lot of "Connection reset" errors that happen twice we should perhaps change "retry once" to "retry five times" or use a more generous delay.

### CODE REVIEW

**Review by daniel-goldstein — APPROVED** [review]

File: hail/python/hailtop/utils/utils.py [hunk 1]
```diff
@@ -799,7 +799,7 @@ async def retry_transient_errors_with_debug_string(debug_string: str, warning_de
             if errors == 1 and is_retry_once_error(e):
                 return await f(*args, **kwargs)
             if not is_transient_error(e):
-                raise
+                raise ValueError(f"errors={errors}, delay={delay}") from e
```
Comment (daniel-goldstein) [inline hail/python/hailtop/utils/utils.py:802]: This smothers the real exception that the caller might expect, e.g. tests that test for a 404

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]

**Comment by daniel-goldstein** [comment]
@danking Is this still relevant? Can we close it since now we're retrying frice?

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
See comment

**Comment by danking** [comment]
debatable. I think it's useful to know this but maybe better conceived as a log message. Let's close and reconsider when its a problem again.

**Pull request closed without merging**